source/mir/random/discrete.d

```diff
@@ -135,9 +137,10 @@ struct Discrete(T)
             arr[sm.index] = AltPair(sm.prob, lg.index);
 
             // update larger & reinsert
-            lg.prob += sm.prob - 1;
+            lg.prob += sm.prob;
+            lg.prob -= 1;
 
-            if (lg.prob < 1)
+            if (lg.prob < 1 || lg.prob.approxEqual(1, 1e-13, 1e-13))
```

Comment:
No need. This is ad-hoc for Tinflex testing and it should not be a part of discrete